ledger/sandbox/src/main/resources/db/migration/postgres/V26_0__Contracts_new_schema.sql

```diff
@@ -0,0 +1,60 @@
+-- Copyright (c) 2019 The DAML Authors. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+---------------------------------------------------------------------------------------------------
+-- V26: Contracts new schema
+--
+-- Used for interpretation and validation by the DAML engine
+---------------------------------------------------------------------------------------------------
+
+-- contains all active and divulged contracts
+create table participant_contracts
+(
+    contract_id varchar primary key not null,
+    template_id varchar not null,
+    -- create_argument needs to be nullable to leave this field empty as part of this migration
+    -- a second (java) migration will fill it up
+    -- a third (sql) migration will make this non-nullable, as it's supposed to always be populated
+    create_argument bytea,
+
+    -- the following fields are null for divulged contracts
+    create_stakeholders varchar array,
+    create_key_hash bytea,
+    create_ledger_effective_time timestamp
+);
+
+-- support looking up a contract by key
+create unique index on participant_contracts(create_key_hash);
+
+insert into participant_contracts
+select
+    contracts.id as contract_id,
+    contracts.package_id || ':' || contracts.name as template_id,
+    null as create_argument, -- filled up in a subsequent migration
+    array_agg(contract_observers.observer) || array_agg(contract_signatories.signatory) as create_stakeholders,
+    decode(contract_keys.value_hash, 'hex') as create_key_hash,
+    ledger_entries.effective_at as create_ledger_effective_time
+from contracts
```

Comment:
I think this should be `from contract_data left join contracts on contract_data.id = contracts.id` (with other necessary changes in the select statement), otherwise this query implicitly filters out the divulged contracts, where there is no entry in `contracts`. The consequence is that we also don't have a `template_id` for those contracts readily available in the database. But this can also be filled in by the java migration via the `ContractInst` that is available in `contract_data`.

Reply:
Actually, I don't think `contract_data` should be there at all. It's not needed in this migration. I'll simply remove it.

Reply:
It's a leftover from my initial query in which I thought I didn't need to go through a Java migration for the step involving `contract_data`.

Reply:
No, it is needed in case of privacy aware ledgers, because then we actually have divulged contracts where there is only an entry in `contract_data` but not in `contracts`.

Reply:
Addressed by https://github.com/digital-asset/daml/pull/5541/commits/4435578f1d993812b3d7c61adb5b590c5481879e